orchagent/port/porthlpr.cpp

```diff
@@ -74,7 +74,8 @@ static const std::unordered_map<std::string, sai_port_fec_mode_t> portFecMap =
 {
     { PORT_FEC_NONE, SAI_PORT_FEC_MODE_NONE },
     { PORT_FEC_RS,   SAI_PORT_FEC_MODE_RS   },
-    { PORT_FEC_FC,   SAI_PORT_FEC_MODE_FC   }
+    { PORT_FEC_FC,   SAI_PORT_FEC_MODE_FC   },
+    { PORT_FEC_AUTO, SAI_PORT_FEC_MODE_NONE }
```

Comment:
@dgsudharsan why fec is None for auto?

Reply:
This is as per the design in the HLD https://github.com/sonic-net/SONiC/pull/1416